doc/model.rst

```diff
@@ -0,0 +1,932 @@
+- `devices_per_node` (int = 1): The number of GPU devices available on the host. This parameter only applies to GPU devices and will be ignored if device is specified as CPU.
+- `first_device` (int = 0): The first GPU device to use on the host. This parameter only applies to GPU devices and will be ignored if device is specified as CPU.
+
+Continue to the :ref:`Loading a TorchScript from File<TS_from_file_ex>` example.
+
+.. _TS_from_file_ex:
+Example: Loading a TorchScript from File
+----------------------------------------
+This example walks through the steps of instructing SmartSim to load a TorchScript from file
+to a ``Orchestrator``.
+
+.. note::
+    This example assumes:
+
+    - a ``Orchestrator`` is launched prior to the ``Models`` execution
+    - an initialized ``Model`` named `smartsim_model` exists within the ``Experiment`` workflow
+
+**Define a TorchScript script**
+
+For the example, we create the create the Python script `torchscript.py`. The file contains a
+simple torch function shown below:
+
+.. code-block:: python
+
+    def negate(x):
+        return torch.neg(x)
+
+**Attach the TorchScript script to a SmartSim Model**
+
+Assuming an initialized ``Model`` named `smartsim_model` exists, we add a TorchScript script using
+the ``Model.add_script()`` function and specify the script path to the parameter `script_path`:
+
+.. code-block:: python
+
+    smartsim_model.add_script(name="example_script", script_path="path/to/torchscript.py", device="GPU", devices_per_node=2, first_device=0)
+
+In the above ``smartsim_model.add_script()`` code snippet, we offer the following arguments:
+
+-  `name` ("example_script"): key to store script under.
+-  `script_path` ("path/to/torchscript.py"): Path to the script file.
+-  `device` ("CPU"): device for script execution.
+-  `devices_per_node` (2): Use two GPUs per node.
+-  `first_device` (0): Start with 0 index GPU.
+
+.. warning::
+    Calling `exp.start(smartsim_model)` prior to instantiation of an orchestrator will result in
+    a failed attempt to load the ML model to a non-existent database.
+
+When `smartsim_model` is started via ``Experiment.start()``, the TorchScript will be loaded from file to the
+orchestrator that is launched prior to the start of the `smartsim_model`.
+
+.. _TS_raw_string:
+---------------------------------
+Define TorchScripts as raw string
+---------------------------------
+Users can upload TorchScript functions from string to send to a colocated or
+standard ``Orchestrator``. This feature is supported by the
+``Model.add_script()`` function's `script` parameter. The function supports
+flexible device selection, allowing users to choose between `"GPU"` or `"CPU"` via the `device` parameter.
+In environments with multiple devices, specific device numbers can be specified using the
+`devices_per_node` parameter.
+
+When specifying a TorchScript using ``Model.add_script()``, the
+following arguments are offered:
+
+- `name` (str): key to store script under.
```

Comment:
key to store function under.

->

reference name for the script inside of the ``Orchestrator``